lib/ddtrace/contrib/active_record/patcher.rb

```diff
@@ -44,7 +44,7 @@ def patch_active_record
         def self.datadog_trace
           # TODO: Consider using patcher for Rails as well.
           # @tracer ||= defined?(::Rails) && ::Rails.configuration.datadog_trace
-          @datadog_trace ||= defined?(::Sinatra) && ::Sinatra::Application.settings.datadog_tracer.cfg
+          @datadog_trace ||= defined?(::Sinatra) && Datadog.configuration[:sinatra].to_h
```

Comment:
It would be interesting if we can avoid this breaking change. Maybe we can use the legacy instrumentation with priority if it's available, falling back to the new one. Consider that the real use case is:
* I have the legacy instrumentation
* I replace it with the new one

It's legit to say "having the legacy instrumentation AND the new one doesn't work fine". Do we have other blockers?

Reply:
@p-lambert ok we can make this breaking change, but it would be great testing in a real application other than relying on tests. Thanks a lot!